pkg/resource/github/deserializer/github_repository_deserializer.go

```diff
@@ -0,0 +1,43 @@
+package deserializer
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/github"
+	"github.com/sirupsen/logrus"
+
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/gocty"
+)
+
+type GithubRepositoryDeserializer struct {
+}
+
+func NewGithubRepositoryDeserializer() *GithubRepositoryDeserializer {
+	return &GithubRepositoryDeserializer{}
+}
+
+func (s GithubRepositoryDeserializer) HandledType() resource.ResourceType {
+	return github.GithubRepositoryResourceType
+}
+
+func (s GithubRepositoryDeserializer) Deserialize(rawResourceList []cty.Value) ([]resource.Resource, error) {
+	resources := make([]resource.Resource, 0)
+	for _, raw := range rawResourceList {
+		raw := raw
+		res, err := decodeRepository(&raw)
+		if err != nil {
+			logrus.Warnf("error when deserializing github_repository %+v : %+v", raw, err)
```

Comment:
I just copy / paste structure from an existing provider, we are doing this everywhere. Maybe we need to refacto this, or maybe not if you are working on a generic deserializer ?

`Wrapf` is more appropriate that `WithMessage` btw as it also add the caller stack.

Reply:
Yep it depends on what we want to do for wrap/message. I removed some log when I refactored supplier but yeah there is still some place where we log and return the same error after. This will probably cause the same log to be shown more than one time.
But yeah you are right it will be fixed on deserializer when the refactoring is done